### PR TITLE
WCOR-23: Typography and base Bootstrap spacing variables

### PR DESCRIPTION
**Pull request by if-nick**
### ODE [255](http://wcorode255.prod.acquia-sites.com/)

**Use/Fix Case** 

- Adds font family, typography sizes and font styling
- Adds overrides for Bootstrap grid spacing/ gutters 
- Adds overrides for Bootstrap margin-bottom on headings, blockquotes, paragraphs and so on 
- Sets Drupal fields, form elements and other HTML elements to use the Westy margin-bottom 

**Implementation Details**

- Most work was completed in `docroot/themes/custom/westy/src/scss/base`
- Also, the Adobe Typekit font was added in `westy.libraries.yml`

**Dependencies**

- The Adobe Typekit font is a temporary package; I.F. had a request that WashCo replace this with their own matching version (see comments below) 

**Test Plan**

- [x] Visit https://wcorode255.prod.acquia-sites.com/animal-services/styles-demo
- [x] https://wcorode255.prod.acquia-sites.com/news?search=Washington
- [x] https://wcorode255.prod.acquia-sites.com/cao/news/moratoria-de-desalojo-en-todo-el-estado
- [x] Verify hyperlinks in page content match the  [Style Tile](https://projects.invisionapp.com/share/R511V1DRTHG7#/screens) color and hover behavior 
- [x] Verify font families match the [Style Tile](https://projects.invisionapp.com/share/R511V1DRTHG7#/screens) 
- [x] Verify font sizes match the [Style Tile](https://projects.invisionapp.com/share/R511V1DRTHG7#/screens) 
- [x] Verify the spacing below block-level elements is `1rem`

### CODE REVIEW

File: docroot/themes/custom/westy/westy.libraries.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@ style:
   version: VERSION
   css:
     theme:
+      //use.typekit.net/axz6wta.css: { type: external, minified: true }
```
Comment (jlwcor) [inline docroot/themes/custom/westy/westy.libraries.yml:5]: Hi @if-nick what is the need to bring in another typekit library?

**Comment by jmilden-git** [comment]
I'll take QA.

**Comment by jmilden-git** [comment]
@if-nick  QA results:
The dark grey (#152E3A) font color should be used for body text and H1-6 tags in the main content area.
The line-height in the masthead looks off.  The H1 - H6 should be reduced from 1.2 to 1
image

image

Search page:  The ODE font is too small.
image
Design comp
image

The hyperlinks styles were not defined in the style tile or style guide.  Look OK, but need Hex value.
Body font is correct family and size, but color should be #152E3A
H1 font:  Should be Acumin Black 45 and dark grey color
H2:  Should be Acumin Light, size is OK
H3: Should be Acumin Black (not Pro), size is OK. Check color value
H4: Should be Acumin Bold, size is OK
H5: Same as H4
H6: Should be Acumin Bold, size is OK
Margin bottom on block elements looks OK at 1rem.

**Comment by if-nick** [comment]
Thanks for this review and comments @jmilden-git. I've made the updates below: 

> The dark grey (#152E3A) font color should be used for body text and H1-6 tags in the main content area.
> The line-height in the masthead looks off. The H1 - H6 should be reduced from 1.2 to 1

For the heading's font families, we should be good. All are in the Acumen-Pro family; but the different font weights are used to specify lights vs bold or black.

**Comment by jmilden-git** [comment]
@if-nick Thanks for those updates, the color looks correct and line height is better.  However now the entire masthead is off.  I know you set the font to H1 as specified in the style tile, but when you do that it doesn't match the dimensions specified in the style guide.
According to the Style Guide
The logo height should be 85px and font-size should be 27px (between h2 - h3).  So we may need a custom style for that masthead title / department text. 
image

**Comment by if-nick** [comment]
>  the color looks correct and line height is better. However now the entire masthead is off.

@jmilden-git , it looks like I'd introduced a regression when updating `page.html.twig`.  That's been updated so the header now has its mobile image and font sizes, as well as desktop image and font sizes and the gutters restored. 

Please let me know if you see any other updates needed

File: docroot/themes/custom/westy/src/scss/base/_typography.scss [hunk 2]
```diff
@@ -1,5 +1,73 @@
...
+h1, h2, h3, h4, h5, h6 {
+  a, a:hover {
+    text-decoration: none;
+  }
+}
+
+h1, h1.page-title {
+  font-size: $h1-mobile-font-size;
+  font-family: acumin-pro, sans-serif;
+  font-weight: 800;
```
Comment (jlwcor) [inline docroot/themes/custom/westy/src/scss/base/_typography.scss:31]: This should be 700

File: docroot/themes/custom/westy/src/scss/base/_typography.scss [hunk 3]
```diff
@@ -1,5 +1,73 @@
...
+h5 {
+  font-family: acumin-pro, sans-serif;
+  font-size: $h5-font-size;
+  font-weight: 700;
+}
+
+h6 {
+  font-family: acumin-pro, sans-serif;
+  font-size: $h6-font-size;
+  font-weight: 600;
```
Comment (jlwcor) [inline docroot/themes/custom/westy/src/scss/base/_typography.scss:65]: This should also be 700

File: docroot/themes/custom/westy/src/scss/base/_typography.scss [hunk 4]
```diff
@@ -1,5 +1,73 @@
...
+  color: map_get($theme-colors, "dark-gray");
+  @include media-breakpoint-up(sm) {
+    font-size: $h1-font-size;
+  }
+}
+
+h2 {
+  font-family: acumin-pro, sans-serif;
+  font-size: $h2-font-size;
+  font-weight: 300;
```
Comment (jlwcor) [inline docroot/themes/custom/westy/src/scss/base/_typography.scss:41]: This should be 300

**Comment by jlwcor** [comment]
@lgbridges @if-nick @jmilden-git based on the style-guide and adobe type-kit recommendations, please note the following on the fonts we are utilizing:

**Acumin Pro Light**

- font-weight is 300
- Only to be used for H2.

**Acumin Pro Regular**
- font-weight is 400
- Only to be used for paragraphs and general body copy.

Acumin Pro Bold
- font-weight is 700
- Only to be used for H1, H3, H4, H5, and H6

**Comment by lgbridges** [comment]
@jlwcor @if-nick @jmilden-git  -- I missed seeing the spec you cite here, but **regardless of what is recommended**, I'm saying 400 weight type does not seem terribly readable.

**Comment by jlwcor** [comment]
> @jlwcor -- I missed seeing the spec you cite here, but **regardless of what is recommended**, I'm saying 400 weight type does not seem terribly readable, at least in this font somehow.

https://projects.invisionapp.com/share/R511V1DRTHG7#/screens 

Body is using Acumin Pro Regular not Acumin Pro Light. Font-weight is specified in Adobe. 

As it stands now in production, not in this PR, it looks pretty readable to me at 400
www washingtoncountyor gov_hhs
.

**Comment by jmilden-git** [comment]
@jlwcor @lgbridges @if-nick  I'm using the style guide and tiles as reference for this QA. 
For body copy I see Acumin-Pro Regular 16 px size.  There is no additional weight specified, this body font should default to "normal".
Using Acumin Pro vs. Acumin is fine, but we have a style guide that calls for multiple variations of font, not different weights of one font.  If we are using what is specified then we won't need to add font-weight.
https://fonts.adobe.com/fonts/acumin#fonts-section

**Comment by jlwcor** [comment]
> Using Acumin Pro vs. Acumin is fine, but we have a style guide that calls for multiple variations of font, not different weights of one font. If we are using what is specified then we won't need to add font-weight.

@jmilden-git again, Adobe specifies what font-weight to use for their fonts. 
Acumin-Pro Regular is supposed to be set at a font-weight of 400.

The weight is what controls what font gets used in the Acumin-Pro suite. Does that make sense?

**Comment by jmilden-git** [comment]
> > Using Acumin Pro vs. Acumin is fine, but we have a style guide that calls for multiple variations of font, not different weights of one font. If we are using what is specified then we won't need to add font-weight.
> 
> @jmilden-git again, Adobe specifies what font-weight to use for their fonts. Acumin-Pro Regular is supposed to be set at a font-weight of 400.
> 
> The weight is what controls what font gets used in the Acumin-Pro suite. Does that make sense?

Yes, I see that now.  However for H1 the style guide is Acumin Black, which Adobe shows as weight 800.

**Comment by jlwcor** [comment]
> Yes, I see that now. However for H1 the style guide is Acumin Black, which Adobe shows as weight 800.

Good catch. That was omitted from the font-family for Acumin pro. Adding it in now.

**Comment by jlwcor** [comment]
@if-nick @jmilden-git @lgbridges 

Current fonts being brought in:

screenshot

**Comment by jmilden-git** [comment]
QA list looks good.

**Comment by if-nick** [comment]
@jmilden-git, @lgbridges, 

The CD is still building but, I believe all changes wanted are in now. The latest change-set makes `main` container content such as paragraphs have 1.25rem bottom margins. And updates the h6 font weight.

**Comment by jlwcor** [comment]
@if-nick not sure how this was missed in QA, but the footer does not seem to be stylized correctly anymore.

**Comment by jlwcor** [comment]
@if-nick looks like 'container' was removed for header nav on default twig. 
Causing issues with horizontal scroll.

screenshot

**Comment by lgbridges** [comment]
@if-nick Any update on this?

**Comment by if-nick** [comment]
> @if-nick Any update on this?

@jmilden-git , @lgbridges , @jlwcor , Team, 

I'm running into an issue with this pull request and it seems to be a limitation of git/  https://github.com. 

I'm wondering if we can move final review of typography to the Search PR, #398?  The ask is because when trying to update `activity/WCOR-23 ` with updates from develop or #398, I have to rebase the branch. Git shows around 5 sets of merge conflicts and each conflict has 3 - 10 files that have to be checked. Unfortunately, in git it happens every time this branch gets a merge/ updated, making it impossible to keep up-to-date. Ideally I'd replace the branch with one that has better history but github isn't allowing me to replace the upstream branch; the PR is tied to that original typography branch. 

All typography/ variables work is included in the search PR. Since we're only checking for regressions at this point, is it possible to do those checks in #398 ? What's best?

**Comment by joshuami** [comment]
> All typography/ variables work is included in the search PR. Since we're only checking for regressions at this point, is it possible to do those checks in #398 ? What's best?

Ultimately, it doesn't matter which PR has the full test plan. It looks like you've put a couple of typography tests in #398, but even a quick review of the CD for that environment still has lots of issues: https://wcorode268.prod.acquia-sites.com/. I'll add specific notes to that PR.

**Pull request closed without merging**